src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+export interface Figure {
+  shape: `triangle` | `circle` | `rectangle`;
+  color: `red` | `green` | `blue`;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a === 0 || b === 0 || c === 0) {
+      throw new Error('Sides can not be equal to zero');
+    }
+
+    if (a + b <= c || c + b <= a || a + c <= b) {
+      throw new Error(
+        'the longest side of a triangle is >= than a sum of two others',
+      );
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: 'circle';
 
-export class Circle implements Figure {}
+  color: 'red' | 'green' | 'blue';
+
+  radius: number;
+
+  constructor(color: 'red' | 'green' | 'blue', radius: number) {
+    if (radius <= 0) {
+      throw new Error('radius can not be equal to zero');
+    }
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  width: number;
+
+  height: number;
+
+  constructor(color: 'red' | 'green' | 'blue', width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('any side can not be equal to zero or less');
+    }
+    this.shape = 'rectangle';
+    this.color = color;
+    this.width = width;
+    this.height = height;
+  }
+
+  getArea(): number {
+    const area = (this.width * this.height * 100) / 100;
+
+    return Math.floor(area * 100) / 100;
```

Comment:
[CODE STYLE] - The `getArea` function should not accept any parameters as all necessary values are already available as object fields. Ensure that the function does not take parameters and uses the class fields directly.